### PR TITLE
Introduce photo-dedup-by-date

### PR DESCRIPTION
**Pull request by aboodman**

### CODE REVIEW

**Comment by zcstarr** [comment]
jenkins: test this

**Comment by zcstarr** [comment]
jenkins: test this

**Comment by zcstarr** [comment]
jenkins: test this

**Comment by zcstarr** [comment]
jenkins: test this

**Comment by zcstarr** [comment]
jenkins: test this

File: samples/go/photo-date-dedup/main.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+}
+
+func index() (win bool) {
+	var dbStr = flag.String("db", "", "input database spec")
+	var outDSStr = flag.String("out-ds", "", "output dataset to write to - if empty, defaults to input dataset")
+	var thresh = flag.Int("threshold", 5000, "Number of milliseconds within which to consider photos duplicates")
+	verbose.RegisterVerboseFlags(flag.CommandLine)
+
+	flag.Usage = usage
+	flag.Parse(false)
```
Comment (willhite) [inline samples/go/photo-date-dedup/main.go]: Why false here? Don't you want to have flexibility in arg order?

File: samples/go/photo-date-dedup/main.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+		})
+		return
+	})
+	flush()
+	close(vals)
+
+	return <-groupBuilder
+}
+
+func makeID() string {
```
Comment (willhite) [inline samples/go/photo-date-dedup/main.go]: We should make a public method somewhere to do this, everyone is going to need it. Also, we should decide if this is a strong enough guarantee for uniqueness.

File: samples/go/photo-date-dedup/main.go [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+			vals <- v
+			group = nil
+		}
+	}
+
+	lastTime := -math.MaxFloat64
+	byDate.IterAll(func(key, s types.Value) {
+		s.(types.Set).IterAll(func(val types.Value) {
+			dt := float64(key.(types.Number))
+			if (dt - lastTime) > float64(thresh*1e6) {
```
Comment (willhite) [inline samples/go/photo-date-dedup/main.go]: Aren't all the dates in this set the same? If so, maybe you could do this check outside the loop for iterating through the set.

**Comment by zcstarr** [comment]
jenkins:test this

**Comment by zcstarr** [comment]
jenkins: test this

**Comment by aboodman** [comment]
jenkins: test this

**Pull request merged**